docs/examples.md

```diff
@@ -0,0 +1,41 @@
+**Selecting the first match via a custom Swift matcher**
+
+When an application has exact duplicate elements (every attribute is the same,
+including their hierarchy), then the correct fix is to update the app to avoid
+duplicating the UI. When that's not possible, a workaround is to match on
+the first element.
+
+```swift
+// Swift Custom Matcher
+/*
+ *  Example Usage:
+ *
+ *  EarlGrey().selectElementWithMatcher(grey_allOfMatchers(
+ *    grey_accessibilityID("some_id"),
+ *    grey_interactable(),
+ *    grey_firstElement())).assertWithMatcher(grey_notNil())
+ *
+ *  Only intended to be used with selectElementWithMatcher. Don't
+ *  use grey_firstElement in assertWithMatcher as it won't do anything.
+ */
+func grey_firstElement() -> GREYMatcher {
+  var firstMatch = true
+  let matches: MatchesBlock = { (element: AnyObject!) -> Bool in
+    if firstMatch {
+      firstMatch = false
+      return true
+    }
+
+    return false
+  }
+
+  let description: DescribeToBlock = { (description: GREYDescription!) -> Void in
+    guard let description = description else {
```

Comment:
ah, i see yes it can be nil...
https://github.com/google/EarlGrey/blob/0481d29a250a26f7281bad2b5b2019a3060a8c82/EarlGrey/Matcher/GREYAnyOf.m#L37

We should fix that.